docs/design/w3c/w3c-representation.md

```diff
@@ -0,0 +1,382 @@
+
+Presentation conversion methods are only need if we decide not to implement duplication flow methods.   
+In this case, library will provide APIs to create/verify Indy formatted presentation and APIs to convert it into/from W3C form.
+
+```rust
+/// Convert Indy styled AnonCreds presentation into W3C AnonCreds presentation form
+///     The conversion process described at the specification: ---
+///
+/// # Params
+/// cred -      object handle pointing to Indy styled AnonCreds presentation to convert
+/// cred_p -    reference that will contain converted presentation (in W3C form) instance pointer
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_presentation_to_w3c(
+    cred: ObjectHandle,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {}
+
+/// Convert W3C styled AnonCreds presentation into Indy styled AnonCreds credential
+///     The conversion process described at the specification: ---
+///
+/// # Params
+/// cred -      object handle pointing to W3C styled AnonCreds presentation to convert
+/// cred_p -    reference that will contain converted presentation (in Indy form) instance pointer
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_presentation_from_w3c(
+    cred: ObjectHandle,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {}
+```
+
+#### Optional: Methods duplication
+
+The idea for this approach to duplicate all issuance/presentation related methods for w3c standard.
+So the credentials and presentations themselves are generated using new flow methods.
+
+> Note, that we still need to implement credential conversion methods to support the case of using existing/issued credentials for doing w3c presentation.
+
+The reasons for adding duplication methods:
+- avoid breaking changes in the existing API 
+  - for example if we want Credential Offer pointing to the form of a credential to be issued
+- clear separation between flows 
+  - if a flow targeting issuing of W3C Credential the specific set of function to be used
+- avoid the situation when function result value may be in different forms
+  - example:
+    - issuer creates offer in Indy form but with indy or w3c indication
+    - as the flow execution result, create credential function returns credential either in w3c or indy form depending on offer
+    - if application analyze credential somehow it cause difficulties
+- easier deprecation of indy styled credentials and APIs
+- presentation conversion methods are not needed anymore in this case
+  - only credential conversion method to do migration for previously issued credentials
+
+```rust
+/// Create Credential Offer according to the AnonCreds specification
+/// It can be either Indy styled or W3C adopted depending on the answer for Q1 
+/// If Indy styled credential to be used, it should indicate that credential in W3C AnonCreds form will be issued as the result.
+///
+/// TO DISCUSS: Should we convert/adopt Credential Offer for W3C form?
```

Comment:
Curious what a W3C form offer looks like.

For AFJ we were intending on superseding RFC 0593, supporting non linked-data signatures, multipe proof types and custom metadata so we can issue AnonCreds credentials (e.g. include cred_def_id and key_proof, etc.. in the proof, as well as have custom properties for the request)

Reply:
As the offer is very dependant on which transport you're using, I don't think AnonCreds spec/lib can define a w3c format.

E.g. the OpenID4VCI has a different way to issue W3C credentials than Aries RFC 0593. The end result is the same: you have a W3C credential with a proof, but the exchange and communication is different.

Reply:
In OpenID4VCs, there is a request-response protocol for issuance. However, I’ve had two people tell me that the required Offer-Request-Issue protocol for AnonCreds issuance is possible using OpenID4VCs.

Reply:
> I’ve had two people tell me that the required Offer-Request-Issue protocol for AnonCreds issuance is possible using OpenID4VCs.

Yes, but it's only possible if we define a custom credential format profile specific to AnonCreds. It can't leverage the existing JSON-LD format defined in the spec AFAIK. Although I'm not sure extra properties are not allowed, so it could be an extension to the JSON-LD format maybe (to allow to pass the AnonCreds offer/request data)